plugins/outputs/influxdb_v2/http.go

```diff
@@ -303,6 +358,124 @@ func (c *httpClient) writeBatch(ctx context.Context, bucket string, metrics []te
 	}
 }
 
+// getOrgID retrieves the organization's ID from its name
+func (c *httpClient) getOrgID(ctx context.Context) (string, error) {
+	loc, err := makeOrgIDURL(*c.url, c.Organization)
+	if err != nil {
+		return "", err
+	}
+
+	req, err := c.makeAPIRequest("GET", loc, nil)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := c.client.Do(req.WithContext(ctx))
+	if err != nil {
+		internal.OnClientError(c.client, err)
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := c.validateResponse(resp.Body)
+	if err != nil {
+		return "", &APIError{
+			StatusCode:  resp.StatusCode,
+			Title:       resp.Status,
+			Description: "An error response was received while attempting to get the organization's ID for: " + c.Organization + ". Error: " + err.Error(),
+		}
+	}
+
+	orgIDResp := &orgIDResponse{}
+	if err := json.NewDecoder(body).Decode(orgIDResp); err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode == 200 && len(orgIDResp.Orgs) == 1 {
+		return orgIDResp.Orgs[0].ID, nil
+	}
+
+	return "", fmt.Errorf("failed to get ID for org %q (do you have org-level read permissions?)", c.Organization)
+}
+
+// CreateBucket creates a new bucket in the configured organization if it doesn't already exist
+func (c *httpClient) CreateBucket(ctx context.Context, bucket string) error {
+	orgID, err := c.getOrgID(ctx)
```

Comment:
The org ID never changes, right? Should we get it only once and reuse it in subsequent calls to CreateBucket?